Common/Source/SaveLoadTask/LoadCupTask.cpp

```diff
@@ -257,104 +257,51 @@ class CupObsZoneUpdater {
-		      // skip TakeOff Set At Home Waypoint
-		      int ix = FindOrAddWaypoint(&(It->second),false);
-		      if (ix>=0) {
-  #if 0 // REMOVE
-			  // We must not change HomeWaypoint without user knowing!
-			  // The takeoff and homewaypoint are independent from task.
-			  // In addition, this is a bug because on next run the index is invalid
-			  // and we have no more HowWaypoint!
-			  HomeWaypoint = ix;
-  #endif
-			  bTakeOff = false;
-		      } else {
-		        BUGSTOP_LKASSERT(0); // .. else is unmanaged, TODO
-		      }
-		  } else {
-
-		    if(Idx < Entries.size())
-		    {
-		      if( _tcscmp(Entries[Idx].c_str(), TpCode) !=0) // doublets?
-		      {
-			int ix =  FindOrAddWaypoint(&(It->second),false);
-			if (ix>=0) Task[idxTP++].Index = ix;
-
-		      }
-		    }
-		  }
-
-	      } else {
-
-		  // An invalid takeoff, probably a "???" , which we ignore
-		  // in any case bTakeOff now is false
-		  bTakeOff=false;
-
-	      }
-	  } else { //ISGAIRRCRAFT
-	      if(It != mapWaypoint.end()) {
-		  if(WPtoAdd!=NULL) {
-		      //add what we found in previous cycle: it was not the last one
-		      int ix = FindOrAddWaypoint(WPtoAdd,false);
-		      if (ix>=0) {
-		        Task[idxTP++].Index = ix;
-		      } else {
-		        BUGSTOP_LKASSERT(0); // .. else is unmanaged, TODO
-		      }
-		      StartupStore(_T("....... bTakeOff: %u %i \n"),(unsigned)(idxTP-1), Task[idxTP-1].Index );
-		  }
-		  if (bTakeOff) { //it's the first: may be we have a corresponding airfield
-		      //look for departure airfield and add it
-		      int ix = FindOrAddWaypoint(&(It->second),true);
-		      if (ix>=0) {
-			  Task[idxTP++].Index = ix;
-			  StartupStore(_T("....... bTakeOff: %u %i \n"),(unsigned)(idxTP-1), Task[idxTP-1].Index );
-			  bTakeOff = false;
-		      } else {
-		        BUGSTOP_LKASSERT(0); // .. else is unmanaged, TODO
-		      }
-		  } else WPtoAdd=&(It->second); //store it for next cycle (may be it is the last one)
-	      }
-	  }
-      } else {
-	  bLoadComplet = false;
-      }
-  }
-  if(ISGAAIRCRAFT) { //For GA: check if we have an airport corresponding to the last WP
-      if(WPtoAdd!=NULL) { //if we have the last one (probably an airfield) still to add...
-	  if(idxTP<MAXTASKPOINTS) {
-	      int ix=FindOrAddWaypoint(WPtoAdd,true); //look for arrival airport and add it
-	      if (ix>=0) {
-		  Task[idxTP++].Index= ix;
-		  StartupStore(_T("....... ISGAAIRCRAFT: %u %i \n"),(unsigned)(idxTP-1), Task[idxTP-1].Index );
-	      } else {
-	        BUGSTOP_LKASSERT(0); // .. else is unmanaged, TODO
-	      }
-	  }
-	  else bLoadComplet=false;
+
+
+    for (uint i =0; i < Entries.size() ; i++)
+    {
+      mapCode2Waypoint_t::iterator It = mapWaypoint.find(Entries[Idx++].c_str());
+      if(It != mapWaypoint.end())
+      {
+        int ix = FindOrAddWaypoint(&(It->second),true);  // first try to find airfield
+        if(ix < 0)
+          ix= FindOrAddWaypoint(&(It->second),true); // not found try if waypoint exist
+
```

Comment:
line 271 and 273 are indentical so 272 and 273 can be removed

Reply:
yes, was a mistake commig from copying already fixed in a new commit.